pipeline/utils.py

```diff
@@ -0,0 +1,56 @@
+import git
+from collections import defaultdict
+
+import sqlalchemy as sa
+
+
+
+def get_image_cache(image_cache=None):
+    if image_cache is None:
+        image_cache = defaultdict(dict)
```

Comment:
What's the benefit of using collections.defaultdict instead of just {} here?

But, also, I think these get_*_cache objects should be combined together into a Context class, as I've mentioned above.

Reply:
The default dict will never say "this key does not exist" so you can just query it using `get` and if it is not there it will be initialized as an empty dict. It's just convenient. 
We can probably do `context` instead.